examples/demo/bytesub/bytesub.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bytesub provides the ability for many clients to subscribe to the
+// latest value of a byte array.
+package bytesub
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+)
+
+type ByteSub struct {
+	nextReady chan struct{}
+	b         []byte
+	r         sync.RWMutex
+}
+
+func New() *ByteSub {
+	return &ByteSub{
+		nextReady: make(chan struct{}),
```

Comment:
I don't think you need this. You can just check if nextReady is nil. That way you can create this struct via `&ByteSub{}`.

Reply:
When a subscriber subscribes before any values have been announced, they need to have a new channel to listen to.  They can't set the value because they only have a read lock.